modules/objdetect/src/face_detect.cpp

```diff
@@ -0,0 +1,242 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+#include "opencv2/imgproc.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/dnn.hpp"
+
+#include <algorithm>
+
+namespace cv
+{
+
+class DNNFaceDetectorImpl : public DNNFaceDetector
+{
+public:
+    DNNFaceDetectorImpl(const String& onnx_path,
+                        const Size& input_size,
+                        float score_threshold = 0.9,
```

Comment:
since it's internal class, do not use default values in the constructor, otherwise you may get some difficult-to-catch bugs; instead, specify all the parameters explicitly inside DNNFaceDetector::create() implementation

Reply:
Done at https://github.com/opencv/opencv/pull/20422/commits/c25681e451a632c0a1eb74c92f800f04251d8640.